test/built-ins/Set/prototype/union/set-like-class-mutation.js

```diff
@@ -0,0 +1,47 @@
+// Copyright (C) 2023 Kevin Gibbons, Anthony Frehner. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+/*---
+esid: sec-set.prototype.union
+description: Set.prototype.union maintains values even when a custom Set-like class mutates the receiver
+features: [Set-methods]
+includes: [compareArray.js]
+---*/
+
+const baseSet = new Set(["a", "b", "c", "d", "e", "q"]);
+
+function mutatingIterator() {
+  let index = 0;
+  let values = ["x", "y"];
+  return {
+    next() {
+      baseSet.delete("b");
+      baseSet.delete("c");
+      baseSet.add("b");
+      baseSet.add("d");
+      return {
+        done: index >= 2,
+        value: values[index++],
+      };
+    },
+  };
+}
+
+const evilSetLike = {
+  size: 2,
+  get has() {
+    baseSet.add("q");
```

Comment:
Should this be
```suggestion
    baseSet.delete("q");
```
?

Reply:
Ah, actually I meant to use `add` but change the expectations for the resulting sets. Not sure how I made that mistake...